(PDB-2955) Decompose open into open and reduce

rbrw opened the pull request.

rbrw committed:
(PDB-2955) Don't call move with replace and atomic

The documentation indicates that atomic is exclusive.  Specifying both
may not hurt anything, but is unnecessary at best.

rbrw committed:
(PDB-2955) Add next-likely-id function

At a minimum, this new public function allows the caller to definitively
partition the entries into those that were stored before this call, and
those that came after.

senior commented on src/stockpile.clj:
```diff
@@ -149,33 +153,21 @@
+(defn next-likely-id
+  "Returns a likely id for the next stored message.  No subsequent
+  entry ids will be less than this value."
+  [q]
```
I think you're going to want a type hint here

rbrw committed:
(PDB-2955) Decompose open into open and reduce

Move open's reduction to a separate public reduce call so that open can
return a queue as soon as it has scanned all the entries.  If desired,
the reducer can refer to the result of a call to next-likely-id to avoid
processing "newer" entries.
